### PR TITLE
Various cleanup & bug fixes

### PR DESCRIPTION
**Pull request by angelaz1**
Doing some small fixes based on feedback from the past few days:

- Disallow rendering of settings and metrics page without proper permissions
- Fixing the white background in the settings page on dark mode
- Input validation on fixed questions

### CODE REVIEW

**Review by jacksontromero — APPROVED** [review]

File: client/src/components/home/student/UpdateQuestionOverlay.tsx [hunk 1]
```diff
@@ -43,18 +43,23 @@ export default function UpdateQuestionOverlay(props) {
...
-          fullWidth
-          sx={{my: 2}}
-          onChange={(event) => setTempQuestion(event.target.value)}
-          inputProps={{maxLength: 256}}
-        />
-
-        <Stack alignItems="center" sx={{pt: 2}}>
-          <Button variant="contained" onClick={printAndClose}>Update Question</Button>
-        </Stack>
+        <form onSubmit={handleSubmit}>
```
[the remark this answers is not in the document]
Comment (amandaLi7) [reply client/src/components/home/student/UpdateQuestionOverlay.tsx:46]: I think this is browser dependent: 
On Chrome, it works fine for me
Screenshot 2023-01-19 at 12 07 02 PM

But on Safari it's different:
Screenshot 2023-01-19 at 12 07 13 PM

**Review by amandaLi7 — APPROVED** [review]
lgtm :))))

File: client/src/contexts/UserDataContext.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import HomeService from '../services/HomeService';
 const UserDataContext = createContext({
   userData: {} as UserData,
   setUserData: ((userData: UserData) => {}) as React.Dispatch<React.SetStateAction<UserData>>,
+  isLoadingUserData: true,
```
[the remark this answers is not in the document]
Comment (amandaLi7) [reply client/src/contexts/UserDataContext.tsx:8]: same here

File: client/src/components/home/student/UpdateQuestionOverlay.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (angelaz1) [reply client/src/components/home/student/UpdateQuestionOverlay.tsx:46]: We love browser differences

File: client/src/contexts/UserDataContext.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (angelaz1) [reply client/src/contexts/UserDataContext.tsx:8]: Yeah I couldn't see the spinner either lol ig it will be there just in case it does take a while

**Pull request merged**